plugins/commands/winrm/command.rb

```diff
@@ -0,0 +1,70 @@
+require 'optparse'
+
+require "vagrant/util/safe_puts"
+
+module VagrantPlugins
+  module CommandWinRM
+    class Command < Vagrant.plugin("2", :command)
+      include Vagrant::Util::SafePuts
+
+      def self.synopsis
+        "executes commands on a machine via WinRM"
+      end
+
+      def execute
+        options = {
+          command: [],
+          shell: :powershell
+        }
+
+        opts = OptionParser.new do |o|
+          o.banner = "Usage: vagrant winrm [options] [name|id]"
+          o.separator ""
+          o.separator "Options:"
+          o.separator ""
+
+          o.on("-c", "--command COMMAND", "Execute a WinRM command directly") do |c|
+            options[:command] << c
+          end
+
+          o.on("-e", "--elevated", "Run with elevated credentials") do |e|
+            options[:elevated] = true
+          end
+
+          o.on("-s", "--shell SHELL", [:powershell, :cmd], "Use specified shell (powershell, cmd)") do |s|
+            options[:shell] = s
+          end
+        end
+
+        argv = parse_options(opts)
+        return if !argv
+
+        with_target_vms(argv, single_target: true) do |machine|
+          if machine.config.vm.communicator != :winrm
+            raise Vagrant::Errors::WinRMInvalidCommunicator,
+              communicator: machine.config.vm.communicator
```

Comment:
This returned nil for me when I executed it against a Ubuntu machine. It ended up reading

> This returned a  communicator.

```
brian@localghost:vagrant-sandbox % be vagrant winrm bork                                                 ±[●][master]
The winrm command requires a WinRM communicator to be used when connecting
to the guest. The guest is currently using a  communicator.
Please update your configuration and try the command again.
```